core/types/l2trace.go

```diff
@@ -145,3 +147,45 @@ type StorageWrapper struct {
 	Key   string `json:"key,omitempty"`
 	Value string `json:"value,omitempty"`
 }
+
+type TransactionData struct {
+	Type     uint8           `json:"type"`
+	Nonce    uint64          `json:"nonce"`
+	TxHash   string          `json:"txHash"`
+	Gas      uint64          `json:"gas"`
+	GasPrice *hexutil.Big    `json:"gasPrice"`
+	From     common.Address  `json:"from"`
+	To       *common.Address `json:"to"`
+	ChainId  *hexutil.Big    `json:"chainId"`
+	Value    *hexutil.Big    `json:"value"`
+	Data     string          `json:"data"`
+	IsCreate bool            `json:"isCreate"`
+	V        *hexutil.Big    `json:"v"`
+	R        *hexutil.Big    `json:"r"`
+	S        *hexutil.Big    `json:"s"`
+}
+
+// NewTraceTransaction returns a transaction that will serialize to the trace
+// representation, with the given location metadata set (if available).
+func NewTraceTransaction(tx *Transaction, blockNumber uint64, config *params.ChainConfig) *TransactionData {
```

Comment:
this function should be `NewTransactionData`.

Reply:
ok, have changed in a [new pr](https://github.com/scroll-tech/go-ethereum/pull/190).